File: mycollab-services/src/main/java/com/mycollab/module/page/domain/Page.java
package com.mycollab.module.page.domain;

import com.mycollab.core.arguments.NotBindable;

import javax.validation.constraints.NotNull;
import java.util.Calendar;

/**
 * @author MyCollab Ltd.
 * @since 4.4.0
 */
public class Page extends PageResource {

    @NotNull(message = "Subject must be not null")
    private String subject;

    @NotNull(message = "Content must be not null")
    private String content;

    @NotBindable
    private boolean isLock = false;

    @NotBindable
    private boolean isNew = true;

    private String category;

    private String status;

    private String lastUpdatedUser;

    @NotBindable
    private Calendar lastUpdatedTime;

    public Page() {
        super();
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public boolean isLock() {
        return isLock;
    }

    public void setLock(boolean isLock) {
        this.isLock = isLock;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public boolean isNew() {
        return isNew;
    }

    public void setNew(boolean isNew) {
        this.isNew = isNew;
    }

    public String getLastUpdatedUser() {
        return lastUpdatedUser;
    }

    public void setLastUpdatedUser(String lastUpdatedUser) {
        this.lastUpdatedUser = lastUpdatedUser;
    }

    public Calendar getLastUpdatedTime() {
        return lastUpdatedTime;
    }

    public void setLastUpdatedTime(Calendar lastUpdatedTime) {
        this.lastUpdatedTime = lastUpdatedTime;
    }
}

File: mycollab-services/src/main/java/com/mycollab/module/page/domain/Page.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
 */
package com.mycollab.module.page.domain

import com.mycollab.core.arguments.NotBindable

import javax.validation.constraints.NotNull
import java.util.Calendar

/**
 * @author MyCollab Ltd.
 * @since 4.4.0
 */
class Page : PageResource() {

    @NotNull(message = "Subject must be not null")
    lateinit var subject: String

    @NotNull(message = "Content must be not null")
    lateinit var content: String

    @NotBindable
    var isLock = false

    @NotBindable
    var isNew = true

    var category: String? = null

    var status: String? = null

    var lastUpdatedUser: String? = null

    @NotBindable
    var lastUpdatedTime: Calendar? = null
}
